rpcs3/Emu/RSX/Overlays/overlay_message_dialog.cpp

```diff
@@ -158,16 +165,58 @@ namespace rsx
 			if (num_progress_bars > 0)
 			{
 				Emu.GetCallbacks().handle_taskbar_progress(0, 1);
+
+				close_requested = 1;
 			}
+			else
+			{
+				fade_animation.current = color4f(1.f);
+				fade_animation.end = color4f(0.f);
+				fade_animation.active = true;
 
-			user_interface::close(use_callback, stop_pad_interception);
+				close_requested = get_system_time() + 150000;
+			}
+			close_use_callback = use_callback;
+			close_stop_pad_interception = stop_pad_interception;
 		}
 
 		struct msg_dialog_thread
 		{
 			static constexpr auto thread_name = "MsgDialog Thread"sv;
 		};
 
+		void message_dialog::update()
+		{
+			const u64 t = get_system_time();
+			if (!num_progress_bars)
+			{
+				if (!creation_time)
+				{
+					// First tick
+					creation_time = t;
+					if (!sound_mute)
```

Comment:
I don't think it makes sense to add this here.
1. why not use it in show ?
2. the game decides if this sound is played, remember? You'll have to set the mute flag to all the places where we create the dialog for rpcs3 reasons.

Btw, why does it use the snd_system_ng instead of snd_error? I know the error doesn't make any sound for some reason, but is it really ng? (I couldn't find an error dialog on my ps3)

Reply:
I don't think 2. was honored.